pkgs/tools/misc/mysqltuner/default.nix

```diff
@@ -14,14 +14,19 @@ stdenv.mkDerivation rec {
   buildInputs = [ perl ];
 
   installPhase = ''
-    mkdir -p $out/bin
-    install -m0755 mysqltuner.pl $out/bin/mysqltuner
+    runHook preInstall
+
+    mkdir -p "$out/bin"
+    install -Dm 0755 mysqltuner.pl "$out/bin/mysqltuner"
+    install -Dm 0644 basic_passwords.txt "$out/bin/basic_passwords.txt"
```

Comment:
You are sure this should go under bin? Isn't share or var more appropriate?

Reply:
This patch could do the trick
```patch
diff --color -ur a/mysqltuner.pl b/mysqltuner.pl
--- a/mysqltuner.pl	2021-02-20 23:56:40.274073201 +0100
+++ b/mysqltuner.pl	2021-02-20 23:58:47.700680798 +0100
@@ -163,7 +163,7 @@
 my $devnull = File::Spec->devnull();
 my $basic_password_files =
   ( $opt{passwordfile} eq "0" )
-  ? abs_path( dirname(__FILE__) ) . "/basic_passwords.txt"
+  ? abs_path( dirname(__FILE__) ) . "/../share/basic_passwords.txt"
   : abs_path( $opt{passwordfile} );
 
 # Username from envvar
```

Reply:
here it's called `$RPM_BUILD_ROOT%{_datarootdir}`

https://github.com/major/MySQLTuner-perl/blob/026ed6db4f6a715d18ef91113cfacba6a5513c52/build/mysqltuner.spec.tpl#L35

Ubuntu has them in

`/usr/share/mysqltuner/basic_passwords.txt`
`/usr/share/mysqltuner/vulnerabilities.csv`

that seem a good place

https://packages.ubuntu.com/groovy/all/mysqltuner/filelist

please also add `vulnerabilities.csv`

Reply:
@SCOTT-HAMILTON are you sure a path is needed? Ubuntu seem not to have any.

https://salsa.debian.org/debian/mysqltuner/-/commit/59f7d66b65c87c8ff84be96e9e994b931d508ba2

if a patch is needed, it would be best to change it upstream

Reply:
Ubuntu doesn't need any because they place install it in LFS dirs, it's this line that make it work on ubuntu and all standard non-nix distros https://github.com/major/MySQLTuner-perl/blob/master/mysqltuner.pl#L183.